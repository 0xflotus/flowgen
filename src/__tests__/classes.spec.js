import {compiler, beautify} from "..";

it("should handle static methods ES6 classes", () => {
  const ts = `class Observable<T> implements Subscribable<T> {
    create: Function;
    static create: Function;
    lift<R>(operator: Operator<T, R>): Observable<R>;
    static lift<R>(operator: Operator<T, R>): Observable<R>;
  }`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
