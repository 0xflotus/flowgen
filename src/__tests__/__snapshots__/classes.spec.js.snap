// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle static methods ES6 classes 1`] = `
"declare class Observable<T> mixins Subscribable<T> {
  create: Function;
  static create: Function;
  lift<R>(operator: Operator<T, R>): Observable<R>;
  static lift<R>(operator: Operator<T, R>): Observable<R>;
}
"
`;
